sktime/classification/distance_based/_time_series_neighbors.py

```diff
@@ -114,46 +111,68 @@ class KNeighborsTimeSeriesClassifier(_KNeighborsClassifier, BaseClassifier):
 
     _tags = {
         "capability:multivariate": True,
+        "X_inner_mtype": ["pd-multiindex", "numpy3D"],
         "classifier_type": "distance",
     }
 
     def __init__(
         self,
         n_neighbors=1,
         weights="uniform",
+        algorithm="brute",
         distance="dtw",
         distance_params=None,
-        algorithm="brute",
+        distance_mtype=None,
         leaf_size=30,
         n_jobs=None,
     ):
         self.n_neighbors = n_neighbors
         self.weights = _check_weights(weights)
+        self.algorithm = algorithm
         self.distance = distance
         self.distance_params = distance_params
-        if isinstance(self.distance, str):
-            distance = distance_factory(metric=self.distance)
-
-        self.algorithm = algorithm
+        self.distance_mtype = distance_mtype
         self.leaf_size = leaf_size
         self.n_jobs = n_jobs
 
-        super(KNeighborsTimeSeriesClassifier, self).__init__(
+        # translate distance strings into distance callables
+        if isinstance(distance, str) and distance not in DISTANCES_SUPPORTED:
+            raise ValueError(
+                f"Unrecognised distance measure string: {distance}. "
+                f"Allowed values for string codes are: {DISTANCES_SUPPORTED}. "
+                "Alternatively, pass a callable distance measure into the constuctor."
+            )
```

Comment:
the comment does not describe what is done. adjust or remove comment

Reply:
how so? I looked at it twice and it seems to match. Please explain. Probably it's sth really silly on my side...

Reply:
i could very well be mistaken, but I think the code directly below the comment just does input check on `distance`.  the str is converted into a function in the separate `_distance` method?

Reply:
ah, you meant *comment* and not *error message*. My bad for misreading what you said. Fixed.